Update to JMeter three

mamhoff opened the pull request:
This 
- gitignores `lock` and `log` files
- version constraints `ruby-jmeter` to `~>3.0`
- removes hardcoding jmeters location on your computer
- changes one method call because its signature been changed in `ruby-jmeter` 3
- adds some more docs

mamhoff committed:
Have git ignore logs and locks

This repo should not have a Gemfile.lock, nor should we accidentally commit logs or jmx files.

mamhoff committed:
Update to JMeter 3.0

JMeter 3.0 has been released, and ruby-jmeter has followed suit.

This requires jmeter 3.0, includes installation instructions for Mac,
and makes the script now agnostic to which minor version of jmeter is
installed in what spot on the machine.

Also adds a version constraint on the `ruby-jmeter` gem because the method
signature for `transactions_per_second` has changed: See
https://github.com/flood-io/ruby-jmeter/commit/95284fa80315948e44c9daf934f4a99eee0aa6fb

cbrunsdon reviewed (approved):
Not 100% sure thats the best way to execute the jmeter, but also good enough for me.

mamhoff commented:
Do not merge yet, as it still has issues actually making orders, which is what I'll hopefully get around to tomorrow.

mamhoff committed:
Use "standard", rather than "default" Cookie policy

I don't know very well what the difference between these two is,
but if I use "standard", Solidus will not give me 401s all the time.

mamhoff committed:
Tweak JSONPath expression and Cookie Manager for JMeter 3

So the previous JSONPath expression for getting the state ID of
New York did not work. I removed the `[0]` for gettting the first
piece of the array and instead rely on jmeter magic variables (`state_id_1`)
for getting the state ID.

Also, the Cookie Manager would not work with the `default` cookie
policy, and the `standard` cookie policy seems to be more like a
real browser.

mamhoff commented:
Now this actually performs orders, and can be merged.
